### PR TITLE
Small fixes to docstrings. Tiny cleanups. Remove an old unused function.

### PR DESCRIPTION
**Pull request by cobordism**
tex_mobject.py - better docstring for remove_stray_braces
mobject.py - fix: "depth" is index 2, not 1
mobject.py - docstring comment correction
mobject.py - digest_mobject_attrs is never used (thankfully)
tex_mobject.py - don't set partent class attributes in child init()

### CODE REVIEW

File: manim/mobject/mobject.py [hunk 1]
```diff
@@ -169,16 +171,17 @@ def remove(self, *mobjects):
...
+    # def digest_mobject_attrs(self):
+    #     """
+    #     Ensures all attributes which are mobjects are included
+    #     in the submobjects list.
+    #     """
+    #     mobject_attrs = [
+    #         x for x in list(self.__dict__.values()) if isinstance(x, Mobject)
+    #     ]
+    #     self.submobjects = list_update(self.submobjects, mobject_attrs)
+    #     return self
```
Comment (leotrs) [inline manim/mobject/mobject.py:184]: Issue: please don't add commented out code to the codebase. If you mean to delete the code, just delete it :slightly_smiling_face: 
```suggestion
```

File: manim/mobject/mobject.py [hunk 2]
```diff
@@ -207,7 +210,7 @@ def copy(self):
         return copy.deepcopy(self)
 
     def generate_target(self, use_deepcopy=False):
-        self.target = None  # Prevent exponential explosion
+        self.target = None  # Prevent unbounded linear recursion
```
Comment (leotrs) [inline manim/mobject/mobject.py:213]: Praise: THANK YOU. I saw this before and it bothered me

**Review by leotrs — CHANGES_REQUESTED** [review]

File: manim/mobject/mobject.py [hunk 3]
```diff
@@ -137,6 +136,9 @@ def add(self, *mobjects):
         return self
 
     def add_to_back(self, *mobjects):
+        """Adds (or moves) all *mobjects to the head of the self.submobjects list.
+        The head of the list is rendered first - i.e. at the back of the scene.
+        """
```
Comment (behackl) [inline manim/mobject/mobject.py:141]: Oneline descriptions should be a bit shorter, technical explanations can come afterwards.

```suggestion
        """Adds (or moves) all passed mobjects to the back of the scene.
        
        .. note::

            Technically, this is done by adding (or moving) the mobjects to
            the head of ``self.submobjects``. The head of this list is rendered
            first, which places the corresponding mobjects behind the
            subsequent list members.
        """
```

File: manim/mobject/svg/tex_mobject.py [hunk 4]
```diff
@@ -311,9 +305,11 @@ def modify_special_strings(self, tex):
         return tex
 
     def remove_stray_braces(self, tex):
+        r"""
+        Makes MathTex resilient to unmatched braces '{'. This is important when the braces in the
+        tex code are spread over multiple agruments eg. MathTex(r"e^{i", "\tau} =1")
```
Comment (behackl) [inline manim/mobject/svg/tex_mobject.py:310]: Similar to above.

```suggestion
        r"""
        Makes :class:`~.MathTex` resilient to unmatched braces.
        
        This is important when the braces in the TeX code are spread over 
        multiple arguments as in, e.g., ``MathTex(r"e^{i", r"\tau} = 1")``.
```

**Review by behackl — COMMENTED** [review]
This LGTM, please consider the two suggestions I've made. (I also pushed a commit running black, I hope you don't mind.)

**Review by leotrs — APPROVED** [review]

**Comment by leotrs** [comment]
Took the liberty of adding @behackl 's changes - one more review and we can merge this

**Review by behackl — APPROVED** [review]
In that case: let's merge after the tests pass.

**Pull request merged**